app/views/leaderboard.scala.html

```diff
@@ -27,9 +27,19 @@ <h5 class="leaderboard-detail">@Messages("leaderboard.overall.detail")</h5>
                         <th class="leaderboard-table-font" scope="row">@{
                             i + 1
                         }</th>
-                        <td class="leaderboard-table-font">@{
-                            stat.asInstanceOf[LeaderboardStat].username
-                        }</td>
+                        @if(user && user.get.role.getOrElse("") != "Anonymous" && stat.asInstanceOf[LeaderboardStat].username == user.get.username){
+                            <td class="leaderboard-table-font">
+                                <a href='@routes.UserProfileController.userProfile(user.get.username)'>
+                                    <u>
+                                        @{stat.asInstanceOf[LeaderboardStat].username} 
+                                    </u>
+                                </a>
+                            </td>
+                        }else{
```

Comment:
Can you try and add spaces around the brackets as per [style guide](https://github.com/ProjectSidewalk/SidewalkWebpage/wiki/Style-Guide) wherever possible in the code you added? I know that template scala prevents you from doing one or two things (like you can't do the `@if (` I would want, you have to do `@if(` instead). But you can do most of it.